# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/InitializedOnce.h
# Commit: e65d7e9dea10
# Full Hash: e65d7e9dea10c417588224abb452aa2d181e8e0c
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-01-13 15:45:24
# Regressor Bug: 1600906
# File Overlap Count: 1
# Description:
#   Bug 1600906 - Convert IDBCursor and BackgroundCursorChild to templates to increase type safety and reduce state. r=dom-workers-and-storage-reviewers,ytausky
#   
#   This also simplifies delegating calls that are dependent on the cursor type.
#   
#   Also reduce dependency on IDBCursor.h by moving enums and type traits to IDBCursorType.h
# ==============================================================================

diff -r 422de232c5ce -r e65d7e9dea10 dom/indexedDB/InitializedOnce.h
--- a/dom/indexedDB/InitializedOnce.h	Fri Jan 10 15:59:51 2020 +0000
+++ b/dom/indexedDB/InitializedOnce.h	Fri Jan 10 15:23:52 2020 +0000
@@ -85,6 +85,10 @@
 
   void reset() {
     MOZ_ASSERT(mMaybe.isSome());
+    maybeReset();
+  }
+
+  void maybeReset() {
     mMaybe.reset();
 #ifdef DEBUG
     mWasReset = true;